### PR TITLE
Update operator development instruction

### PR DESCRIPTION
**Pull request by tedhtchang**
## Why are these changes needed?

Update local operator development instruction

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by tedhtchang** [comment]
@kevin85421 please take a look. Thanks.

File: ray-operator/DEVELOPMENT.md [hunk 1]
```diff
@@ -79,6 +79,22 @@ kubectl logs {YOUR_OPERATOR_POD} | grep "Hello KubeRay"
 * The command `make docker-build` (Step 3) will also run `make test` (unit tests).
 * Step 6 also installs the custom resource definitions (CRDs) used by the KubeRay operator.
 
+### Start kuberay-operator locally
```
Comment (kevin85421) [inline ray-operator/DEVELOPMENT.md:82]: We do not encourage contributors to develop KubeRay outside of the k8s cluster.

File: ray-operator/DEVELOPMENT.md [hunk 2]
```diff
@@ -120,29 +136,29 @@ IMG=kuberay/operator:nightly make deploy
...
+# v0.5.0 (go1.19)
 ```
 
-Make sure your `go` version is still 1.17:
+Make sure your `go` version is still 1.19:
 
 ```bash
 go version
-# go version go1.17.13 darwin/arm64
+# go version go1.19 darwin/arm64
```
Comment (kevin85421) [inline ray-operator/DEVELOPMENT.md:158]: KubeRay currently does not support ARM 64 well.

File: ray-operator/DEVELOPMENT.md (hunk 1 above)
Comment (tedhtchang) [reply ray-operator/DEVELOPMENT.md]: Got it. I removed the instruction.

File: ray-operator/Makefile [hunk 3]
```diff
@@ -81,8 +81,8 @@ build: generate fmt vet ## Build manager binary.
     "                                           \
     -o bin/manager main.go
 
-run: manifests generate fmt vet ## Run a controller from your host.
-	go run ./main.go
+run: manifests generate fmt vet ## Run the controller from your host.
```
Comment (kevin85421) [inline ray-operator/Makefile:84]: Why do we need this in case the KubeRay operator has to communicate with the Ray head when using RayJob or RayService?

Comment (tedhtchang) [reply ray-operator/Makefile:84]: You mean why we need to do `make run` ? 
It starts the kuberay operator locally as if it were deployed to a cluster. I think basically this lets you re-built and re-start the operator quickly.

Comment (kevin85421) [reply ray-operator/Makefile:84]: > KubeRay operator has to communicate with the Ray head when using RayJob or RayService?

KubeRay operator needs to send HTTP requests to the Ray head when using RayJob or RayService. How does KubeRay communicates with the head Pod If we run the operator locally?

Comment (tedhtchang) [reply ray-operator/Makefile:84]: good question. I guess this is the reason you don't want people run operator outside the of a cluster. maybe I should remove this instead.

Comment (kevin85421) [reply ray-operator/Makefile:84]: > I guess this is the reason you don't want people run operator outside the of a cluster.

Yes. Maybe we can remove it from the Makefile.

Comment (tedhtchang) [reply ray-operator/Makefile:84]: I will leave a note and commented it out in case someone wants to enable it.

Comment (kevin85421) [reply ray-operator/Makefile]: It's fine, but I may delete it in the future.

**Review by kevin85421 — APPROVED** [review]

**Pull request merged**